x/cosmwasm/bindings/types.go

```diff
@@ -0,0 +1,32 @@
+package cosmwasm
+
+type Swap struct {
+	PoolId   int    `json:"pool_id"`
+	DenomIn  string `json:"denom_in"`
+	DenomOut string `json:"denom_out"`
+}
+
+type Step struct {
+	PoolId   int    `json:"pool_id"`
+	DenomOut string `json:"denom_out"`
+}
+
+type SwapAmount struct {
+	In  string `json:"in,omitempty"`
```

Comment:
I think we can use sdk.Int here (which is used by sdk.Coin and will parse the json string into an integer for in-memory usage).

I am not sure how that works with `omitempty`, maybe it needs to be `*sdk.Int`?

Could you change the type and add some tests for serialization and parsing?

Reply:
I will.